fix(dev-infra): properly tag github releases as prerelease as appropriate

josephperrott opened the pull request:
Previously all github releases created by the release tooling tagged
releases as `latest`. Instead releases which are created for the `next`
tag on NPM should be tagged as `prerelease` for github.

gkalpak commented on dev-infra/release/publish/actions.ts:
```diff
@@ -494,9 +496,10 @@ export abstract class ReleaseAction {
    * @param newVersion The new version to be published.
    * @param publishBranch Name of the branch that contains the new version.
    * @param npmDistTag NPM dist tag where the version should be published to.
+   * @param prerelease Whether the release is a prerelease, i.e. a next version.
```
Aren't RCs prereleases too?

```suggestion
   * @param prerelease Whether the release is a prerelease, i.e. a next or RC version.
```

Or did you mean "a version tagged as 'next' on NPM" (in which case, why not use `npmDistTag` to compute `prerelease` :grin:)?

gkalpak commented on dev-infra/release/publish/actions.ts:
```diff
@@ -494,9 +496,10 @@ export abstract class ReleaseAction {
    */
   protected async buildAndPublish(
-      newVersion: semver.SemVer, publishBranch: string, npmDistTag: string) {
+      newVersion: semver.SemVer, publishBranch: string, npmDistTag: string, prerelease: boolean) {
```
Why not use `npmDistTag !== 'next'` to determine whether it is a prerelease or not?
What would be the downsides? Is there any scenario where this check would not work?

josephperrott replied:
Yeah, I initially thought of doing it that way but was worried that I was missing something that prevented us from doing this.  But if that is your thought as well, lets go with doing this check instead.

gkalpak reviewed (approved):
:+1: (Just one question/suggestion.)

josephperrott committed:
fix(dev-infra): properly tag github releases as prerelease as appropriate

Previously all github releases created by the release tooling tagged
releases as `latest`. Instead releases which are created for the `next`
tag on NPM should be tagged as `prerelease` for github.
